### PR TITLE
Exaplin the failure

### PR DESCRIPTION
**Pull request by olivergondza**
I am observing

```
hudson.util.HudsonFailedToLoad: org.jvnet.hudson.reactor.ReactorException: java.lang.Error: java.lang.reflect.InvocationTargetException
	at hudson.WebAppMain$3.run(WebAppMain.java:237)
Caused by: org.jvnet.hudson.reactor.ReactorException: java.lang.Error: java.lang.reflect.InvocationTargetException
	at org.jvnet.hudson.reactor.Reactor.execute(Reactor.java:269)
	at jenkins.InitReactorRunner.run(InitReactorRunner.java:44)
	at jenkins.model.Jenkins.executeReactor(Jenkins.java:937)
	at jenkins.model.Jenkins.<init>(Jenkins.java:818)
	at hudson.model.Hudson.<init>(Hudson.java:83)
	at hudson.model.Hudson.<init>(Hudson.java:79)
	at hudson.WebAppMain$3.run(WebAppMain.java:225)
Caused by: java.lang.Error: java.lang.reflect.InvocationTargetException
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:110)
	at hudson.init.TaskMethodFinder$TaskImpl.run(TaskMethodFinder.java:176)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:282)
	at jenkins.model.Jenkins$8.runTask(Jenkins.java:926)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:210)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:106)
	... 8 more
Caused by: java.lang.NullPointerException
	at com.redhat.jenkins.plugins.ci.messaging.MessageProviderMigrator.migrateCIMessageBuilders(MessageProviderMigrator.java:135)
	... 13 more
```

caused by the fact the plugin this is branched from is present as well. This is how the exception looks like now:

```
hudson.util.HudsonFailedToLoad: org.jvnet.hudson.reactor.ReactorException: java.lang.Error: java.lang.reflect.InvocationTargetException
	at hudson.WebAppMain$3.run(WebAppMain.java:237)
Caused by: org.jvnet.hudson.reactor.ReactorException: java.lang.Error: java.lang.reflect.InvocationTargetException
	at org.jvnet.hudson.reactor.Reactor.execute(Reactor.java:269)
	at jenkins.InitReactorRunner.run(InitReactorRunner.java:44)
	at jenkins.model.Jenkins.executeReactor(Jenkins.java:937)
	at jenkins.model.Jenkins.<init>(Jenkins.java:818)
	at hudson.model.Hudson.<init>(Hudson.java:83)
	at hudson.model.Hudson.<init>(Hudson.java:79)
	at hudson.WebAppMain$3.run(WebAppMain.java:225)
Caused by: java.lang.Error: java.lang.reflect.InvocationTargetException
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:110)
	at hudson.init.TaskMethodFinder$TaskImpl.run(TaskMethodFinder.java:176)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:282)
	at jenkins.model.Jenkins$8.runTask(Jenkins.java:926)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:210)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:106)
	... 8 more
Caused by: java.lang.AssertionError: Version mismatch: GlobalCIConfiguration provided by other plugin: Plugin:redhat-ci-plugin
	at com.redhat.jenkins.plugins.ci.GlobalCIConfiguration.get(GlobalCIConfiguration.java:188)
	at com.redhat.jenkins.plugins.ci.messaging.MessageProviderMigrator.migrateCIMessageBuilders(MessageProviderMigrator.java:135)
	... 13 more

```

### CODE REVIEW

File: src/main/java/com/redhat/jenkins/plugins/ci/GlobalCIConfiguration.java [hunk 1]
```diff
@@ -82,6 +87,7 @@ public GlobalCIConfiguration(List<JMSMessagingProvider> configs) {
         this.configs = configs;
     }
 
+    @DataBoundConstructor
```
Comment (olivergondza) [inline src/main/java/com/redhat/jenkins/plugins/ci/GlobalCIConfiguration.java]: Not sure if ever used, but `@DataBoundSetter` is reported not to work without `@DataBoundConstructor`.

File: src/main/java/com/redhat/jenkins/plugins/ci/GlobalCIConfiguration.java [hunk 2]
```diff
@@ -46,21 +53,20 @@
...
     private transient String broker;
     private transient String topic;
     private transient String user;
     private transient Secret password;
     private transient boolean migrationInProgress = false;
 
-    private static final String PLUGIN_NAME = Messages.PluginName();
 
-    public static final GlobalCIConfiguration EMPTY_CONFIG =
-            new GlobalCIConfiguration(Collections.<JMSMessagingProvider>emptyList());
```
Comment (olivergondza) [inline src/main/java/com/redhat/jenkins/plugins/ci/GlobalCIConfiguration.java]: Unused.

File: src/main/java/com/redhat/jenkins/plugins/ci/GlobalCIConfiguration.java [hunk 3]
```diff
@@ -168,17 +174,22 @@ public boolean configure(StaplerRequest req, JSONObject json) throws hudson.mode
...
+        return Messages.PluginName();
     }
 
-
-    public static GlobalCIConfiguration get() {
-        try {
-            return (GlobalConfiguration.all() == null ? null : GlobalConfiguration.all().get(GlobalCIConfiguration.class));
-        } catch (Exception e) {
-            log.warning("Unable to get global configuration.");
+    public static @Nonnull GlobalCIConfiguration get() {
```
Comment (olivergondza) [inline src/main/java/com/redhat/jenkins/plugins/ci/GlobalCIConfiguration.java]: Every client of this API relies it never returns null...

**Comment by scoheb** [comment]
Looks good to me! The test error seems to be random. I have another PR that is coming that may fix it.
Thanks!

**Pull request merged**